cypress/e2e/gui/testsGui.cy.js

```diff
@@ -0,0 +1,358 @@
+      cy.intercept(
+        'GET',
+        `${CUSTOMERS_API_URL}?**`,
+        {
+          delay: 1000,
+          fixture: 'customers'
+        }
+      ).as('getCustomers');
+      cy.setCookie('cookieConsent', 'accepted');
+      cy.visit('/');
+    });
+    it('Check the load element', () => {
+      cy.get('#loading').should('be.visible');
+    });
+  });
+
+  context('Validate the customer positive scenarios', () => {
+    beforeEach(() => {
+      const CUSTOMERS_API_URL = `${Cypress.env('API_URL')}/customers`;
+      cy.intercept(
+        'GET',
+        `${CUSTOMERS_API_URL}?**`,
+        { fixture: 'customers' }
+      ).as('getCustomers');
+      cy.setCookie('cookieConsent', 'accepted');
+      cy.visit('/');
+    });
+    context('Theme scenarios', () => {
+      it('Change to dark mode', () => {
+        cy.get('[aria-label="theme light activated"]').click();
+
+        cy.get('[aria-label="theme dark activated"]').should('be.visible');
+      });
+      it('Return to light mode', () => {
+        cy.get('[aria-label="theme light activated"]').click();
+        cy.get('[aria-label="theme dark activated"]').click();
+
+        cy.get('[aria-label="theme light activated"]').should('be.visible');
+      });
+    });
+
+    it('Check limit on localstorage', () => {
+      cy.get('[name="pagination-limit"]').select('50');
+      cy.getAllLocalStorage().then((localStorage) => {
+        const { paginationLimit } = localStorage['http://localhost:3000'];
+        expect(paginationLimit).to.equal('50');
+      });
+    });
+
+    it('Download CSV', () => {
+      cy.contains('button', 'Download CSV').click();
+      cy.readFile('cypress/resource/customersReference.csv', 'utf-8').then((csvReference) => {
+        cy.readFile('cypress/downloads/customers.csv', 'utf-8').then((csvDownload) => {
+          expect(csvDownload).to.equal(csvReference);
+        });
+      });
+    });
+    it('Show the customer address', () => {
+      cy.contains('button', 'View').click();
+      cy.contains('Show address').click();
+
+      cy.contains('14135 Kari Garden Suite 427').should('be.visible');
+      cy.contains('Mooreshire').should('be.visible');
+      cy.contains('Nevada').should('be.visible');
+      cy.contains('64043').should('be.visible');
+      cy.contains('United States of America').should('be.visible');
+    });
+    it('Hide the customer address', () => {
+      cy.contains('button', 'View').click();
+      cy.contains('Show address').click();
+
+      cy.contains('Hide address').click();
+      cy.contains('Show address').should('be.visible');
+    });
+    it('Apply a filter, go to a customer page, return to the list page, and check if the previous filters are selected', () => {
+      cy.get('[data-testid="size-filter"]').select('Medium');
+      cy.get('[data-testid="industry-filter"]').select('Retail');
+      cy.wait('@getCustomers');
+      cy.get('[aria-label="View company: Littel Co"]').click();
```

Comment:
Que tal algo mais simples, como o seguinte?

```suggestion
      cy.contains('button', 'View').click();
```